internal/worker.go

```diff
@@ -251,6 +251,12 @@ type (
 		// Note: Cannot be enabled at the same time as EnableSessionWorker
 		UseBuildIDForVersioning bool
 
+		// Optional: Assign a Deployment Name to this worker, an identifier for Worker Versioning that
+		// groups task queues for the given BuildID.
+		// NOTE: Experimental
+		// Note: Both BuildID and UseBuildIDForVersioning need to also be set to enable the new Worker Versioning-3 feature.
```

Comment:
Nit using `Note` and `NOTE` in the same doc we should be consistent here

Can we validate this note and fail if `DeploymentName` is set but `BuildID` or `UseBuildIDForVersioning` is not set?

Reply:
Thanks. I think sometimes we may want to provide a Deployment Name even if we have `UseBuildIDForVersioning=false`, the same as we can currently provide the BuildID without forcing versioning. For example, when turning versioning on dynamically with an env property, or using it with visibility queries, or with pollers info...